tools/create_github_release.py

```diff
@@ -0,0 +1,90 @@
+#!/usr/bin/env python
+"""
+Post-release script to download artifacts from azure pipelines and use them to create
+a GitHub release.
+
+Setup:
+ - Create an azure personal access token
+   - https://docs.microsoft.com/en-us/azure/devops/organizations/accounts/use-personal-access-tokens-to-authenticate
+   - You'll need read scope
+   - Save the token to somewhere like ~/.ssh/azurepat.txt
+ - Create a github personal access token
+   - https://docs.github.com/en/github/authenticating-to-github/creating-a-personal-access-token#creating-a-token
+   - You'll need repo scope
+   - Save the token to somewhere like ~/.ssh/githubpat.txt
+
+Run:
+
+python tools/create_github_release.py 1.7.0 ~/.ssh/azurepat.txt ~/.ssh/githubpat.txt
+"""
+
+import requests
+import sys
+import tempfile
+import urllib.parse
+from zipfile import ZipFile
+
+from azure.devops.connection import Connection
+from github import Github
+from msrest.authentication import BasicAuthentication
+
+def download_azure_artifacts(azure_access_token, tempdir):
+    """Download and unzip build artifacts from Azure pipelines.
+
+    :param str azure_access_token: string containing azure access token
+    :param str path: path to a temporary directory to save the files
+
+    """
+    # Create a connection to the azure org
+    organization_url = 'https://dev.azure.com/certbot'
+    credentials = BasicAuthentication('', azure_access_token)
+    connection = Connection(base_url=organization_url, creds=credentials)
+
+    # Find the build artifacts
+    build_client = connection.clients.get_build_client()
+    get_builds_response = build_client.get_builds('certbot', definitions='3')
+    build_id = get_builds_response.value[0].id
+    artifacts = build_client.get_artifacts('certbot', build_id)
+
+    # Save and unzip files
+    for filename in ('windows-installer', 'changelog'):
+        url = build_client.get_artifact('certbot', build_id, filename).resource.download_url
+        r = requests.get(url)
+        with open(tempdir + '/' + filename + '.zip', 'wb') as f:
+            f.write(r.content)
+        with ZipFile(tempdir + '/' + filename + '.zip', 'r') as zipObj:
+           zipObj.extractall(tempdir)
+
+def create_github_release(github_access_token, tempdir, version):
+    """Download and unzip build artifacts from Azure pipelines.
+
+    :param str github_access_token: string containing github access token
+    :param str path: path to a temporary directory where azure artifacts are located
+    :param str version: Certbot version number, e.g. 1.7.0
+
+    """
+    # Create release
+    g = Github(github_access_token)
+    repo = g.get_user('certbot').get_repo('certbot')
+    release_notes = open(tempdir + '/changelog/release_notes.md', 'r').read()
+    release= repo.create_git_release('v{0}'.format(version),
+                                     'Certbot {0}'.format(version),
+                                     release_notes)
```

Comment:
> Today I learned that with GitHub's API creating a release and adding assets to it are two different steps unlike what we've been doing manually.

They sure are! In no version of the api that I could find does it allow you to upload an asset on creation.

Ah, I didn't know we were planning to do that. I'm much more comfortable with the draft version, actually, especially since in addition, that way if the upload fails we won't have a partially updated release. Thanks for finding this solution!

Reply:
Plus a fun bonus from testing! If the release already exists, it fails at the "update to not-a-draft" stage. So you can run the script to check everything without even manually changing the version number to an old release (which is what I did to check that everything completes), and it'll just run up until there so you can check manually on the website that the info and assets are there.